gitops-v2/new/push.yaml

```diff
@@ -71,38 +82,94 @@ jobs:
-
-            ARTIFACT_PATH="${WORKSPACE}/${SERVICE_NAME}/${ARTIFACT_FILE_NAME}"
-            if [[ ! -f "${ARTIFACT_PATH}" ]]
-            then
-                echo "Missing artifact file ($ARTIFACT_PATH) from CI pipeline, did you trigger this pipeline manually?"
-                exit 1
-            fi
+
             IMAGE_ID_FULL="${ACR_NAME}.azurecr.io/${IMAGE_PATH_PREFIX}/${IMAGE_ID}"
-            az acr login -n "${ACR_NAME}"
-            docker load < $ARTIFACT_PATH
+            # Get token and store it in variables
+            az acr login -n "${ACR_NAME}" -t > tmp.json
+            TOKEN=$(cat tmp.json |jq -r .accessToken)
+            SERVER=$(cat tmp.json |jq -r .loginServer)
+
+            echo "##vso[task.setvariable variable=token]$TOKEN"
+            echo "##vso[task.setvariable variable=server]$SERVER"
+        condition: ne('${{ parameters.azureSubscription }}', '')
+
+      - task: AmazonWebServices.aws-vsts-tools.AWSShellScript.AWSShellScript@1
+        displayName: "Get ECR login info"
+        inputs:
+          workingDirectory: "."
+          awsCredentials: ${{ parameters.awsServiceConnection }}
+          regionName: ${{ parameters.awsRegion }}
+          scriptType: inline
+          inlineScript: |
+            set -e
+
+            aws ecr describe-repositories --repository-names ${SERVICE_NAME} || aws ecr create-repository --repository-name ${SERVICE_NAME} --image-scanning-configuration scanOnPush=true
+            SERVER=$(aws ecr describe-repositories --repository-names ${SERVICE_NAME} | jq -r '.repositories[0].repositoryUri')
+            TOKEN=$(aws ecr get-login-password --region ${AWS_REGION})
+
+            echo "##vso[task.setvariable variable=token]$TOKEN"
+            echo "##vso[task.setvariable variable=server]$SERVER"
+        env:
+          SERVICE_NAME: $(resources.triggeringAlias)
+        condition: ne('${{ parameters.awsRegion }}', '')
+
+      - bash: |
+          set -e
+          WORKSPACE=$(Pipeline.Workspace)
+          EXTRA_TAG=${{ parameters.extraTag }}
+          SHA_SHORT=$(shaShort)
+          TOKEN=$(token)
+          SERVER=$(server)
+          ACR_NAME=${{ parameters.acrName }}
+          IMAGE_PATH_PREFIX=${{ parameters.imagePathPrefix }}
+          IMAGE_ID_FULL="${ACR_NAME}.azurecr.io/${IMAGE_PATH_PREFIX}/${IMAGE_ID}"
+
+          ARTIFACT_PATH="${WORKSPACE}/${SERVICE_NAME}/${ARTIFACT_FILE_NAME}"
+          if [[ ! -f "${ARTIFACT_PATH}" ]]
+          then
+              echo "Missing artifact file ($ARTIFACT_PATH) from CI pipeline, did you trigger this pipeline manually?"
+              exit 1
+          fi
+
+          docker load < $ARTIFACT_PATH
+
+          # Run AWS
+          if [[ ${{ parameters.awsRegion }} != "" ]]
+          then
+            docker login -u AWS -p $TOKEN $SERVER
+            docker tag $(imageId) ${SERVER}:${SHA_SHORT}
+            docker push ${SERVER}:${SHA_SHORT}
+          fi
+
+          # Run Azure
+          if [[ $IMAGE_PATH_PREFIX != "" ]]
```

Comment:
The reason why I use IMAGE_PATH_PREFIX is that I got issues with the templating things from main.yaml so I just took a variable that would always be a simple string.

Same thing with awsRegion.